packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -490,9 +496,14 @@ Future<void> _buildGradleProjectV2(
       final Map<String, dynamic> manifest = <String, dynamic>{
         'baselineChecksum': baselineChecksum,
         'buildNumber': package.versionCode,
-        'patchNumber': buildInfo.patchNumber,
       };
 
+      if (buildInfo.patchNumber != null) {
+        manifest.addAll(<String, dynamic>{
```

Comment:
nit: `manifest['patchNumber'] = ...` might be more clear since we're adding just one key-value.